api4/user.go

```diff
@@ -1952,7 +1960,14 @@ func loginCWS(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 	c.LogAuditWithUserId(user.Id, "success")
 	c.App.AttachSessionCookies(c.AppContext, w, r)
-	http.Redirect(w, r, *c.App.Config().ServiceSettings.SiteURL, http.StatusFound)
+
+	redirectURL := *c.App.Config().ServiceSettings.SiteURL
+	if campaign != "" {
+		if url, ok := campaignToURL[campaign]; ok {
+			redirectURL += url
+		}
+	}
```

Comment:
This would be a good opportunity to emit a telemetry event. May be worth while looping in Don or Sri on if they want anything specifically

Reply:
+1, created ticket at: https://mattermost.atlassian.net/browse/MM-48759